devicetypes/smartthings/zigbee-multi-switch-power.src/zigbee-multi-switch-power.groovy

```diff
@@ -0,0 +1,183 @@
+				attributeState "turningOff", label: '${name}', action: "switch.on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+			}
+			tileAttribute("power", key: "SECONDARY_CONTROL") {
+				attributeState "power", label: '${currentValue} W'
+			}
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main "switch"
+		details(["switch", "refresh", "power"])
+	}
+}
+
+def installed() {
+	createChildDevices()
+	updateDataValue("onOff", "catchall")
+}
+
+def updated() {
+	log.debug "updated()"
+	updateDataValue("onOff", "catchall")
+}
+
+def parse(String description) {
+	Map eventMap = zigbee.getEvent(description)
+	Map eventDescMap = zigbee.parseDescriptionAsMap(description)
+
+	if (!eventMap && eventDescMap) {
+		eventMap = [:]
+		if (eventDescMap?.clusterId == zigbee.ONOFF_CLUSTER) {
+			eventMap[name] = "switch"
+			eventMap[value] = eventDescMap?.value
+		}
+	}
+
+	if (eventMap) {
+		if (eventDescMap?.sourceEndpoint == "01" || eventDescMap?.endpoint == "01") {
+			sendEvent(eventMap)
+		} else {
+			def childDevice = childDevices.find {
+				it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.sourceEndpoint}" || it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.endpoint}"
+			}
+			if (childDevice) {
+				childDevice.sendEvent(eventMap)
+			} else {
+				log.debug "Child device: $device.deviceNetworkId:${eventDescMap.sourceEndpoint} was not found"
+			}
+		}
+	}
+}
+
+private void createChildDevices() {
+	def i = 2
+	addChildDevice("Child Switch Health Power", "${device.deviceNetworkId}:0${i}", device.hubId,
+		[completedSetup: true, label: "${device.displayName[0..-2]}${i}", isComponent: false])
+}
+
+private getChildEndpoint(String dni) {
+	dni.split(":")[-1] as Integer
+}
+
+def on() {
+	log.debug("on")
+	zigbee.on()
+}
+
+def off() {
+	log.debug("off")
+	zigbee.off()
+}
+
+def childOn(String dni) {
+	log.debug(" child on ${dni}")
+	def childEndpoint = getChildEndpoint(dni)
+	zigbee.command(zigbee.ONOFF_CLUSTER, 0x01, "", [destEndpoint: childEndpoint])
+}
+
+def childOff(String dni) {
+	log.debug(" child off ${dni}")
+	def childEndpoint = getChildEndpoint(dni)
+	zigbee.command(zigbee.ONOFF_CLUSTER, 0x00, "", [destEndpoint: childEndpoint])
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	return refresh()
+}
+
+def refresh() {
+	zigbee.onOffRefresh() +
+	zigbee.electricMeasurementPowerRefresh()
+	zigbee.readAttribute(zigbee.ONOFF_CLUSTER, 0x0000, [destEndpoint: 2]) +
+	zigbee.readAttribute(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER, 0x050B, [destEndpoint: 2]) +
+}
+
+def poll() {
+	refresh()
+}
+
+def healthPoll() {
+	log.debug "healthPoll()"
+	def cmds = refresh()
+	cmds.each { sendHubCommand(new physicalgraph.device.HubAction(it)) }
+}
+
+def configureHealthCheck() {
```

Comment:
Most of this code is deprecated at this point. 

Just set the checkInterval to 12 minutes and have the `ping` method poll both devices.

The healthPoll and all that is handled by Device Health now.

Reply:
ok